### PR TITLE
Entry point redirection and default routers configuration

### LINKED ISSUE #4863: Traefik 2.0 verbosity in setting up HTTP to HTTPS redirects
One thing that confused me a bit when trying to set up a 2.0 test was the verbosity of supporting both http and https but with redirects:

```yml
  whoami:
    image: containous/whoami
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.whoami_insecure.rule=Host(`whoami.domain.com`)"
      - "traefik.http.routers.whoami_insecure.middlewares=file.https-redirect"
      - "traefik.http.routers.whoami.rule=Host(`whoami.domain.com`)"
      - "traefik.http.routers.whoami.tls=true"
```

```toml
[EntryPoints]
  [EntryPoints.HTTP]
    address = ":80"

  [EntryPoints.HTTPS]
    address = ":443"

[Providers]
  [Providers.Docker]
    Watch = true
    ExposedByDefault = false

  [Providers.File]

[http.middlewares]
  [http.middlewares.https-redirect.redirectscheme]
    scheme = "https"
```

With the current syntax I'd have to repeat that _insecure route for each service. It would be nice if there was a way to statically configure that as a default in traefik.toml, similar to the old way of adding a redirect to an entrypoint.

Thanks!

### PR DESCRIPTION
**Pull request by ldez**
### What does this PR do?

Introduce a simple way to:
- create global entry point redirection (ex: HTTP to HTTPS)
- set a default routers configuration by entry point.

<details>
<summary>redirection definition on entry points</summary>

```yml
## Static configuration
entryPoints:
  web:
    address: ":80"
    http:
      redirections:
        entryPoint:
          to: websecure
          scheme: https

  websecure:
    address: ":443"
```

</details>

<details>
<summary>default routers configuration</summary>

```yml
## Static configuration
entryPoints:
  web:
    address: ":80"

  websecure:
    address: ":443"
    http:
      middlewares: # default middlewares
        - foobar
        - foobar
      tls:  # default TLS configuration
        options: foobar
        certResolver: foobar
        domains:
          - main: foo
            sans:
              - foo
              - foo
          - main: bar
            sans:
              - bar
              - bar
```

</details>

### Motivation

Related to #5630, #6280
Fixes #4863, #5645

### More

- [x] Added/updated tests
- [x] Added/updated documentation

### Additional Notes

Co-authored-by: Julien Salleyron <julien.salleyron@gmail.com>
Co-authored-by: Mathieu Lonjaret <mathieu.lonjaret@gmail.com>

### CODE REVIEW

File: pkg/server/aggregator_test.go [hunk 1]
```diff
@@ -381,8 +390,202 @@ func TestAggregator_tlsStore(t *testing.T) {
...
+				HTTP: &dynamic.HTTPConfiguration{
+					Routers:     make(map[string]*dynamic.Router),
+					Middlewares: make(map[string]*dynamic.Middleware),
+					Services:    make(map[string]*dynamic.Service),
+					Models:      make(map[string]*dynamic.Model),
+				},
+			},
+		},
+		{
+			desc: "with model, no used",
```
Comment (mpl) [inline pkg/server/aggregator_test.go]:
```suggestion
			desc: "with model, not used",
```

File: pkg/server/aggregator_test.go [hunk 2]
```diff
@@ -381,8 +390,202 @@ func TestAggregator_tlsStore(t *testing.T) {
...
+						"websecure@internal": {
+							Middlewares: []string{"test"},
+							TLS:         &dynamic.RouterTLSConfig{CertResolver: "ep"},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "with model, two entry point",
```
Comment (mpl) [inline pkg/server/aggregator_test.go]:
```suggestion
			desc: "with model, two entry points",
```

File: docs/content/routing/entrypoints.md [hunk 3]
```diff
@@ -529,3 +529,211 @@ If the Proxy Protocol header is passed, then the version is determined automatic
...
+
+??? info "`entryPoint.to`"
+    
+    _Required_
+    
+    The target entry point.
+
+    ```toml tab="File (TOML)"
+    [entryPoints.foo]
+      # ...
```
[the remark this answers is not in the document]
Comment (ldez) [reply docs/content/routing/entrypoints.md]: yes because the entry point need `address`

**Comment by GustavoAmerico** [comment]
The domains attribute is required ?

**Comment by ldez** [comment]
no field is required for TLS

**Comment by sshishov** [comment]
How it is supposed to work, for instance, with Helm Chart V2 (Kubernetes) which you are maintaining. HTTPS entrypoint there is on port `8443`. When I enable redirect on `web` to `websecure` and specify `scheme=https`, the redirect becomes `http://host.example.com:8443` which of cause cannot be reached as the service and LB is listening on port `433`?

**Comment by Sarke** [comment]
I just wanted to say "Thanks!", this saves sooo much in the config files.

Just `--entrypoints.websecure.http.tls=true` reduces the amount of config labels by half.

**Pull request merged**